drivers: regulator: mspm0: Add supoort for TI MSPM0 VREF module

Girinandha-M opened the pull request:
The TI MSPM0 internal VREF module is primarily used as a reference voltage source for the integrated ADC and comparator (COMP) peripherals. 
Add driver support for TI MSPM0 VREF module. 

Signed-off-by: Girinandha Manivelpandiyan <girinandha@linumiz.com>

Girinandha-M committed:
dts: arm: ti: mspm0: Add VREF node for TI MSPM0

Add dts support for VREF Node for internal reference voltage.

Signed-off-by: Girinandha Manivelpandiyan <girinandha@linumiz.com>

Girinandha-M committed:
dts: bindings: regulator: Add bindings for TI MSPM0 VREF module

Add bindings for TI MSPM0 VREF module.

Signed-off-by: Girinandha Manivelpandiyan <girinandha@linumiz.com>

Girinandha-M committed:
drivers: regulator: mspm0: Add supoort for TI MSPM0 VREF module

TI MSPM0 VREF module is used by analog peripherals for reference voltage
Add support for TI MSPM0 VREF module.

Signed-off-by: Girinandha Manivelpandiyan <girinandha@linumiz.com>

parthitce commented on drivers/regulator/Kconfig.mspm0:
```diff
@@ -0,0 +1,18 @@
+
+config REGULATOR_MSPM0_VREF_INIT_PRIORITY
+	int "MSPM0 VREF peripheral driver init priority"
+	default 45
```
Any specific reason for 45?

parthitce commented on include/zephyr/dt-bindings/regulator/mspm0_vref.h:
```diff
@@ -0,0 +1,27 @@
+ * @name MSPM0 VREF Regulator API Modes
+ * @{
+ */
+#define MSPM0_VREF_MODE_NORMAL 0
```
if this is for dts, then IMO this change needs to be part bindings commit?

parthitce commented on drivers/regulator/regulator_mspm0_vref.c:
```diff
@@ -0,0 +1,266 @@
+{
+	const struct regulator_mspm0_vref_config *config = dev->config;
+
+	DL_VREF_enableInternalRef(config->regs);
```
`dev->config->regs` directly works

parthitce commented on drivers/regulator/regulator_mspm0_vref.c:
```diff
@@ -0,0 +1,266 @@
+{
+	const struct regulator_mspm0_vref_config *config = dev->config;
+
+	DL_VREF_disableInternalRef(config->regs);
```
`dev->config->regs` directly works

tbursztyka replied:
Unless enabling and disabling HAL calls are atomic, you might want to protect it if CONFIG_REGULATOR_THREAD_SAFE_REFCNT is enabled

tbursztyka commented on drivers/regulator/regulator_mspm0_vref.c:
```diff
@@ -0,0 +1,266 @@
+
+	if (common->refcnt != 0) {
+		volt_get = (config->regs->CTL0 & VREF_CTL0_BUFCONFIG_MASK) ? VREF_1_4V : VREF_2_5V;
+
```
you can remove this empty line  as your are testing below the assignment you made above.

tbursztyka commented on drivers/regulator/regulator_mspm0_vref.c:
```diff
@@ -0,0 +1,266 @@
+	if (common->refcnt != 0) {
+		mode_get = (config->regs->CTL0 & VREF_CTL0_SHMODE_MASK) ? MSPM0_VREF_MODE_SHMODE
+			    : MSPM0_VREF_MODE_NORMAL;
+
```
same here

Girinandha-M committed:
boards: ti: lp_mspm0g3507: enable VREF support

enable VREF peripheral support and add vref to the list
of supported hardware in board yaml

Signed-off-by: Girinandha Manivelpandiyan <girinandha@linumiz.com>

Girinandha-M committed:
boards: ti: lp_mspm0g3519: enable VREF support

enable VREF peripheral and add vref to the list of
supported hardware in board yaml

Signed-off-by: Girinandha Manivelpandiyan <girinandha@linumiz.com>

Girinandha-M committed:
boards: ti: lp_mspm0l2228: enable VREF support

enable VREF peripheral and add vref to the list of
supported hardware in board yaml

Signed-off-by: Girinandha Manivelpandiyan <girinandha@linumiz.com>

parthitce commented on dts/bindings/regulator/ti,mspm0-vref.yaml:
```diff
@@ -0,0 +1,47 @@
+
+properties:
+  reg:
+    required: true
```
new line after each property

parthitce commented on drivers/regulator/regulator_mspm0_vref.c:
```diff
@@ -0,0 +1,263 @@
+
+LOG_MODULE_REGISTER(vref, CONFIG_LOG_DEFAULT_LEVEL);
+
+#define VREF_1_4V	1400000
```
micro volts?
